### DISCUSSION TITLE
Getting chart information from pygwalker.

### DISCUSSION THREAD

**Discussion by YcZen** [Kanaries/pygwalker#614] [Q&A] [answered]
**Hi, Pygwalker community!** 

**Does anyone have an idea how to get chart information from Pygwalker, particularly the names of the x and y axes? Please see the picture below. I have another program that needs the chart information as input. We know charts change dynamically. I guess there exists a way to get the chart information whenever it updates. I have tried a lot of approaches, but all failed.**
![image](https://github.com/user-attachments/assets/84f3a0ee-e5d9-421b-9a75-3624b9cfd218)

 **Ideally, _my program should be able to dynamically get the information in vis_spec (see figure below), or the corresponding JSON data directly._ It has everything I need. Using Pygwalker.vis_spec does not work, and it always returns an empty list. I really appreciate it if someone could offer me a solution.** 
![image](https://github.com/user-attachments/assets/59b83386-05f7-4786-8024-db9c3445170e)

**Comment by longxiaofei** [comment]
Hi @YcZen, In which application are you using pygwalker? jupyter? streamlit?

**Reply by YcZen** [reply]
Hi@longxiaofei, thanks for asking! I am using streamlit.

**Reply by longxiaofei** [reply, marked as answer]
You can refer to this issue to complete your needs currently.

https://github.com/Kanaries/pygwalker/issues/580

**Reply by YcZen** [reply]
> You can refer to this issue to complete your needs currently.
> 
> #580

oh... I have been struggling to solve this issue for days. Didn't expect this simplicity. Thank you! You are very helpful!
